src/main/java/org/recommendation/service/GenreTopItemFinder.java

```diff
@@ -0,0 +1,74 @@
+package org.recommendation.service;
+
+import org.recommendation.model.Genre;
+import org.recommendation.model.Movie;
+import org.recommendation.model.User;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.recommendation.data.helper.MovieDataHelper.movieMap;
+import static org.recommendation.data.helper.UserDataHelper.userMap;
+
+public class GenreTopItemFinder extends TopItemFinder {
```

Comment:
Interface is missing for this GenreTopItemFinder class

Reply:
added